intelmq/lib/utils.py

```diff
@@ -714,6 +718,36 @@ def send(self, *args, **kwargs):
         return super().send(*args, **kwargs)
 
 
+def create_request_session() -> requests.Session:
```

Comment:
Looks similar to `intelmq.lib.bot.Bot.set_request_parameters` + `intelmq.lib.utils.create_request_session_from_bot`. Can we reduce the duplication a bit? The `set_request_parameters` could also be used statically I guess.

Reply:
The deeper reasoning behind this is explained above.

The code of `update_database` covers rare cases when there are more than one of the same experts in the pipeline. But with such consideration the code doesn't know which one's configuration (for proxy) to use. To make it predictable, it only uses `defaults.conf`. And for the class method approach (explained above) there is no method to load `requests.Session` object only based on `defaults.conf`. I don't see how to use `intelmq.lib.utils.create_request_session_from_bot` statically, it uses `self` for pretty much everything. :-(

Reply:
I have updated the function `intelmq.lib.utils.create_request_session_from_bot` to a more universal approach (and renamed to reflect the change). Now it is optional to pass the bot instance to the function. If no bot is specified, only values from `defaults.conf`. Also I came across #1594.